pkg/cluster/actions.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+
+	"sigs.k8s.io/kind/pkg/cluster/config"
+)
+
+// Action define a set of tasks to be executed on a `kind` cluster.
+// Usage of actions allows to define repetitive, high level abstractions/workflows
+// by composing lower level tasks
+type Action interface {
+	// Tasks returns the list of task that are identified by this action
+	// Please note that the order of task is important, and it will be
+	// respected during execution
+	Tasks() []Task
+}
```

Comment:
i guess the list of actions is not exposed on the user side yet - i.e. we expose the Node list in the config separated by `---`, but i didn't see a way to also provide a list of actions?

Reply:
You are right. UX around actions is not part of this PR.

With regards to exposing actions via Config/Api objects in a future, nothing in this PR prevent this, but IMO the decision mostly depends by the answer to this thread https://github.com/kubernetes-sigs/kind/issues/161#issuecomment-445802956.